// sass-brunch/lib/index.js
(function() {
  var SassCompiler, spawn, sysPath;

  spawn = require('child_process').spawn;

  sysPath = require('path');

  module.exports = SassCompiler = (function() {

    SassCompiler.prototype.brunchPlugin = true;

    SassCompiler.prototype.type = 'stylesheet';

    SassCompiler.prototype.extension = 'scss';

    SassCompiler.prototype.pattern = /\.s[ac]ss$/;

    function SassCompiler(config) {
      this.config = config;
      null;
    }

    SassCompiler.prototype.compile = function(data, path, callback) {
      var error, options, result, sass;
      result = '';
      error = null;
      options = ['--stdin', '--load-path', sysPath.join(this.config.rootPath, 'app', 'styles'), '--no-cache'];
      if (/\.scss$/.test(path)) options.push('--scss');
      sass = spawn('sass', options);
      sass.stdin.end(data);
      sass.stdout.on('data', function(stdout) {
        return result = stdout;
      });
      sass.stderr.on('data', function(stderr) {
        return error = stderr;
      });
      return sass.on('exit', function(code) {
        return callback(error, result.toString());
      });
    };

    return SassCompiler;

  })();

}).call(this);
